### ISSUE TITLE
Slider and CountController do not update on rebuild page setting

### ISSUE THREAD

**Issue by ericwiedemann** [FlutterFlow/flutterflow-issues#2021]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

Ok ,

Let me state that I do have a workaround built but the workaround breaks other functionality on the page specifically a timer. I am also trying to find a workaround for the timer and I think I have it but just need more time coding it.

Problem:

I have a slider and a countcontroller on a page.
Both of these are linked by an app state variable that is set for each widget's initial value. 

When I increase or decrease the CountController. I define an action to update the app state and set a variable's value from the CountController Value, I select the Rebuild Current Page setting in the action.

It sets the app state variable value but does not rebuild (reload) the slider widget with the current variable value on the page.

When I define an action to set the sliders value to the variable, I convert the value to an integer and set the app state variable value I then select Rebuild Current Page. It sets the variable value but does not rebuild the countercontroller widget with the current variable value on the page.

### Expected Behavior

When the Slider value is changed the CountController Value is updated in the ui. 

When the CountController Value is changed the Slider Value is updated ui.

This does not happen the page will rebuild and set the App State Variable but the ui will not change.

When using a text widget to debug setting it to the variable, the text widget will update but not the slider or the CountController.

### Create A New Project
**1. Setup widgets**
Set an _App State Variable_ **SliderValue**

Create A **Slider Widget**
Set _Show Value_ and _Round Decimal_ to **True**
Set _Time Delay_ to **500**

Create A **CountController Widget** (Counter Button)

Set the initial value of _CountController_ and _Slider_ to _App State Variable_ **SliderValue**.

Create a Text Field and have the text output be the _App State Variable_ **SliderValue**

**2. Create String to Integer Function and set Variable**
Select _Slider_ and set an action _Update App State_ select value to be **SliderValue**
in the value to set : _create a new custom function_. 
**convertStringtoInt**
```
import 'dart:convert';
import 'dart:math' as math;

import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';
import 'package:intl/intl.dart';
import 'package:timeago/timeago.dart' as timeago;
import '/flutter_flow/lat_lng.dart';
import '/flutter_flow/place.dart';
import '/flutter_flow/uploaded_file.dart';
import '/flutter_flow/custom_functions.dart';

int convertStringtoInt(String? slider1values) {
  /// MODIFY CODE ONLY BELOW THIS LINE

  // convert string to integer the argument is slider1values
  return int.parse(slider1values ?? '0');

  /// MODIFY CODE ONLY ABOVE THIS LINE
}
```
Set the Custom Function _Return Value_ to an integer.
Set _Arguments_ to Slider1Values to be a string.

Return to Action Editor and set the value to be set to _Custom Function_ **convertStringtoInt**
Set _slider1values_ argument to be the widget state of the **slider**.

**3. Set Action on CountController**
Select CountController and set and action _Update App State_ 
Select value to be updated to be **SliderValue**
Set Value to be Widget State CountController.

4. Run App in Test Mode
Change the **slider** and the text value of the text will change but not the **countcontroller**

Change the **countcontroller** and the text value of the text will change but the slider will not change its value.

### 5. Create Workaround
1. Duplicate the existing widgets.
2. Set an additional action on both the slider and counter to Navigate to the current page.

Test app 
Functionality works as expected but the app is jumpy and will break other widgets like timers if on the page.

### Reproducible from Blank

- [x] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT8ehcn15ZR2rsRE0JD+asdVmSQsQzN+a7ssltYbGA0dB5DoPJk+Y87/N05pTcOmTXx2JU+Whnoy08rNv8zuMvJdaA6rfN1+w89XEDnJRmGtMIiNGJWKXnJROJlVIG6v0qunvRR/OvduZXAl72amJtm5SSTZJbL1NjIBRciaBLnXj3iQAhzBJRRGujQDRlTXC/QHZgaMkaDR2adU3ezrxw==

### Context

I want to be able to rebuild the page and have the values update properly and they do not.

The workaround that I built works but it breaks the seamlessness of the app and makes it jumpy and also breaks other functionality like a timer running on the page.

### Visual documentation

you can download / copy / view the app example here.

https://app.flutterflow.io/project/slider-value-count-controller-example-sjldks

### Additional Info

I have Developed a workaround:

In each action workflow for both the slider and the countcontroller I define a navigate to action and select the current page, I must also select allow back navigation, If I do not select allow back navigation the widgets do not update. With the allow back navigation selected they work as expected.

The issue with the workaround:

When I navigate to the current page with back navigation the timer on the page resets and stops running.

Attempts to make a workaround for this secondary issue was to stop the timer, save timer value and then reload the current timer value on page load and start timer if timer was running but I am running into a few issues there as well.

The best solution would be to have the expected functionality of rebuilding the current page setting rebuild each countcontroller and slider widgets from the value set.

Please let me know if I am just ignorant on how best to do this or I have discovered an actual bug, as well as a solution to such issue.

Update:

I have found another issue/characteristic of this when I continued testing the app. When I launch the app in phone view the slider is hidden due to being hidden on the phone. If I change the value of the countcontroller and then change the view to ipad the slider value will be changed to the stored variable value but once the slider has been viewed, if I change back to the phone view and try to use the countcontroller to change values it will not continue to change the slider value. This only works if the slider has not been loaded into view before.

### Environment

```markdown
- FlutterFlow version: current
- Platform: flutterflow testing evnvironment app
- Browser name and version: chrome
- Operating system and version affected: macOS ventura
```

**Comment by hariprasadms** [comment]
Hi @ericwiedemann - Are you still facing the same issue ?
I just tested it using both app state and page states, all works fine. see below. 

Here is the sample project to clone - https://app.flutterflow.io/project/slider-value-count-controller-example-1ywjpq

https://github.com/FlutterFlow/flutterflow-issues/assets/17435678/c582d173-7400-48bd-9ca5-db69c4847beb

**Comment by ericwiedemann** [comment]
Very nice example but it seems you skipped a few steps and misunderstood the problem. 

The Slider and the Controller should share the **same** app state or page state variable.

I have the convert string to int function, because the slider value is double/string and not an int. 

That is why it is included in the instructions.

Would you like me to modify your example or do you understand the difference in your example and mine?

**Comment by ericwiedemann** [comment]
I went ahead and made the changes to your app so you can see what I meant

**Comment by ericwiedemann** [comment]
I have fixed the issue. I am documenting it so if others have a similar issue they can know the solution.

In the workflow for the Slider you must also define an action to Set Form Field and choose the Counter, and set the value by converting the widget state slider value to int using the custom function.
In the workflow for the Counter you must also define an action to Set Form Field and choose the Slider, and set the value to the widget state counter value.

I have updated the app so it works properly now.
If you would like to clone and keep it live as an example for others feel free to do so. as I will probably delete it in a few weeks or so.
